pkg/apis/core/validation/conditional_validation.go

```diff
@@ -17,24 +17,73 @@ limitations under the License.
 package validation
 
 import (
+	"fmt"
+	"net"
+	"strings"
+
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/features"
+	netutil "k8s.io/utils/net"
 )
 
 // ValidateConditionalService validates conditionally valid fields.
-func ValidateConditionalService(service, oldService *api.Service) field.ErrorList {
+func ValidateConditionalService(service, oldService *api.Service, allowedIPFamilies ...api.IPFamily) field.ErrorList {
```

Comment:
making this a vararg makes it too easy to call this without realizing this is an option and accidentally allow too much

Reply:
ok